packages/core/src/utils/indyIdentifiers.ts

```diff
@@ -0,0 +1,114 @@
+import type { CredentialDefinitionTemplate, SchemaTemplate } from '../modules/ledger/services/IndyLedgerService'
+import type { CredDef, Schema } from 'indy-sdk'
+
+import { generateCredentialDefinitionId, generateSchemaId } from '../modules/ledger/ledgerUtil'
+
+import { didFromCredentialDefinitionId, didFromSchemaId } from './did'
+
+/**
+ *
+ * @see For the definitions below see also: https://hyperledger.github.io/indy-did-method/#indy-did-method-identifiers
+ *
+ */
+export type Did = 'did'
+export type DidIndyMethod = 'indy'
+// Maybe this can be typed more strictly than string. Choosing string for now as this can be eg just `sovrin` or eg `sovrin:staging`
+export type DidIndyNamespace = string
+//  NOTE: because of the ambiguous nature - whether there is a colon or not within DidIndyNamespace this is the substring after the ***last*** colon
+export type NamespaceIdentifier = string
+
+// TODO: This template literal type can possibly be improved. This version leaves the substrings as potentially undefined
+export type IndyNamespace = `${Did}:${DidIndyMethod}:${DidIndyNamespace}:${NamespaceIdentifier}`
+
+export function isQualifiedIndyIdentifier(identifier: string | undefined): boolean {
+  if (!identifier || identifier === '') return false
+  return identifier.startsWith('did:indy:')
+}
+
+export function getLegacyIndyCredentialDefinitionId(qualifiedIdentifier: string) {
+  if (!isQualifiedIndyIdentifier(qualifiedIdentifier)) return qualifiedIdentifier
```

Comment:
should we throw an error here? Or do you think it it better to just silently continue?

Reply:
Also rather unsure. Gut feeling is silently continue but I do see the benfit of throwing an error